File: Two Pointer/reverseString.js
const reverseString = (s) => {
  let left = 0;
  let right = s.length - 1;

  while (left < right) {
    [s[left], s[right]] = [s[right], s[left]];
    left++;
    right--;
  }
};

// Time complexity: O(n), where n is the length of the input array.
// We need to traverse half of the array to reverse it.
// Space complexity: O(1)

File: Two Pointer/reverse_string.py
from typing import List


class ReverseString:
    def reverseString(self, s: List[str]) -> None:
        left = 0
        right = len(s) - 1
        while left < right:
            s[left], s[right] = s[right], s[left]
            left += 1
            right -= 1

# Time complexity: O(n), where n is the length of the input list. 
# We need to traverse half of the list to reverse it.
# Space complexity: O(1)